parser/encoding/request.go

```diff
@@ -0,0 +1,138 @@
+package encoding
+
+import (
+	"github.com/cockroachdb/errors"
+
+	"encr.dev/parser"
+	v1 "encr.dev/proto/encore/parser/schema/v1"
+)
+
+// ParamLocation is the request/response home of the parameter
+type ParamLocation int16
+
+const (
+	Header = 1 // Parameter is placed in the http header
+	Query  = 2 // Parameter is placed in the query string
+	Body   = 3 // Parameter is placed in the body
+)
+
+// tagToLocation is used to translate struct field tags to param locations
+// if overrideDefault is set, the default http param location will be changed
+// if the tag matches the paramLocation, the param name will be override with the
+// tag name
+var tagToLocation = map[string]struct {
+	location        ParamLocation
+	overrideDefault bool
+}{
+	"query": {
+		location:        Query,
+		overrideDefault: true,
+	},
+	"qs": {
+		location:        Query,
+		overrideDefault: true,
+	},
+	"header": {
+		location:        Header,
+		overrideDefault: true,
+	},
+	"json": {
+		location:        Body,
+		overrideDefault: false,
+	},
+}
+
+// RequestEncoding expresses how a request should be encoded for a specified set of HttpMethods
+type RequestEncoding struct {
+	// The HTTP methods these field configurations can be used for
+	HttpMethods []string
```

Comment:
In Go terms like HTTP and ID should be all uppercase (`HTTP`, `ID`, not `Http`/`Id`), so `HTTPMethods`